tests/func/test_scm.py

```diff
@@ -44,3 +46,76 @@ def test_lfs_prefetch(tmp_dir, dvc, scm, mocker):
     with dvc.switch(rev):
         lfs_prefetch(dvc.dvcfs, ["foo"])
         mock_fetch.assert_called_once()
+
+
+def test_download_lfs_prefetch_directory(tmp_dir, dvc, scm, mocker):
+    mock_fetch = mocker.patch("scmrepo.git.lfs.fetch")
+    tmp_dir.scm_gen(
+        {
+            ".gitattributes": "data/**/* filter=lfs diff=lfs merge=lfs -text",
+            "data/test.txt": "test data",
+        },
+        commit="init lfs",
+    )
+    rev = scm.get_rev()
+    with dvc.switch(rev):
+        download(dvc.dvcfs, "data", "data")
+        mock_fetch.assert_called_once_with(
+            scm, [rev], include=["/data/**"], progress=ANY
+        )
+
+
+def test_download_lfs_prefetch_file(tmp_dir, dvc, scm, mocker):
+    mock_fetch = mocker.patch("scmrepo.git.lfs.fetch")
+    tmp_dir.scm_gen(
+        {
+            ".gitattributes": "data/**/* filter=lfs diff=lfs merge=lfs -text",
+            "data/test.txt": "test data",
+        },
+        commit="init lfs",
+    )
+    rev = scm.get_rev()
+    with dvc.switch(rev):
+        download(dvc.dvcfs, "data/test.txt", "data")
+        mock_fetch.assert_called_once_with(
+            scm, [rev], include=["/data/test.txt"], progress=ANY
+        )
+
+
+@pytest.mark.skipif(os.name == "nt", reason="invalid directory name syntax on Windows")
```

Comment:
are any of special (to be escaped) symbols allowed on NT? (make a simpler test that runs there as well?)

Reply:
Good point. I've checked [Microsoft docs](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions) and the relevant characters are `*` and `?`. I've refactored the tests to test each subpattern case independently and skip only those on Windows that are incompatible. WDYT?